src/client/cockpit-beiboot

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import asyncio
+import base64
+import importlib.resources
+import json
+import logging
+import lzma
+import os
+import shlex
+import sys
+from pathlib import Path
+from typing import Iterable, Optional, Sequence
+
+from cockpit import polyfills
+from cockpit._vendor import ferny
+from cockpit._vendor.bei import beipack, bootloader
+from cockpit.bridge import setup_logging
+from cockpit.channel import ChannelRoutingRule
+from cockpit.channels import PackagesChannel
+from cockpit.data import read_cockpit_data_file
+from cockpit.packages import Package, Packages
+from cockpit.peer import Peer
+from cockpit.protocol import CockpitProblem
+from cockpit.router import Router, RoutingRule
+from cockpit.transports import StdioTransport
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_ferny_askpass() -> Path:
+    src_path = importlib.resources.files(ferny.__name__) / 'interaction_client.py'
+    src_data = src_path.read_text()
+
+    xdg_data_home = os.environ.get('XDG_DATA_HOME')
+    if xdg_data_home is None:
+        xdg_data_home = os.path.expanduser('~/.local/share')
+    dest_path = Path(xdg_data_home, 'cockpit-client-askpass')
+
+    logger.debug("Checking if %s exists...", dest_path)
+
+    # Check first to see if we already wrote the current version
+    try:
+        if dest_path.read_text() != src_data:
+            logger.debug("  ... it exists but is not the same version...")
+            raise ValueError
+        if not dest_path.stat().st_mode & 0o100:
+            logger.debug("  ... it has the correct contents, but is not executable...")
+            raise ValueError
```

Comment:
When would that happen accidentally? IMHO we should either not touch an existing file at all, or always chown it for simplicity.

Reply:
It's just a consistency check.  I can't imagine this happening by "accident", but the executable bit is essential, so I double-check it.

I'm trying to avoid modifying the filesystem if I don't need to.

I already rewrote this a couple of times, and I agree that it's a bit awkward how it is.  Do you have a better suggestion?

Reply:
It's fine, I mostly wanted to check if you had a legit case in mind.